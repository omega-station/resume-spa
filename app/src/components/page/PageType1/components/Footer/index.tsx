import React from 'react';
import Contact from '../../../../core/Contact';
import StyledFooter from './style';

const Footer = (): JSX.Element => {
  return (
    <StyledFooter>
      <Contact hasDescription={true} isPostalWithCity={true} />
    </StyledFooter>
  );
};

export default Footer;
