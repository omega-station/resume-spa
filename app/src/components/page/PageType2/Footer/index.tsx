import React from 'react';
import { coding, reading, ucDigger, ucMain } from '../../../../images';
import StyledFooter from './style';

const Footer = (): JSX.Element => {
  const url = 'https://en.wikipedia.org/wiki/Webring';

  const counter = (
    <div>
      <a href="https://www.hitwebcounter.com/" target="_blank" rel="noopener noreferrer">
        <img src="https://hitwebcounter.com/counter/counter.php?page=7200063&style=0038&nbdigits=7&type=page&initCount=325145" alt="hitwebcounter" />
      </a>
    </div>
  );

  return (
    <StyledFooter>
      <div>
        <div>
          <div>
            <img src={coding} alt="" />
          </div>
          <div>
            <p>
              This{' '}
              <a href={url} target="_blank" rel="noopener noreferrer">
                Amazing Front-end Developer's Ring
              </a>{' '}
              site is owned by{' '}
              <a href={url} target="_blank" rel="noopener noreferrer">
                Paul Kevin Koehler
              </a>
              .
            </p>
            <p>
              Want to join the{' '}
              <a href={url} target="_blank" rel="noopener noreferrer">
                Amazing Front-end Developer's Ring
              </a>
              ?
            </p>
          </div>
          <div>
            <img src={reading} alt="" />
          </div>
        </div>
        <div>
          [
          <a href={url} target="_blank" rel="noopener noreferrer">
            Prev
          </a>
          ] [
          <a href={url} target="_blank" rel="noopener noreferrer">
            Next
          </a>
          ] [
          <a href={url} target="_blank" rel="noopener noreferrer">
            Random
          </a>
          ] [
          <a href={url} target="_blank" rel="noopener noreferrer">
            List
          </a>
          ]
        </div>
      </div>
      <div>
        <img src={ucDigger} alt="" />
        <img src={ucMain} alt="" />
        <img src={ucDigger} alt="" />
      </div>
      {counter}
    </StyledFooter>
  );
};

export default Footer;
