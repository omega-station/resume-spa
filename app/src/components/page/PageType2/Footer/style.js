import styled from '@emotion/styled';
import { border } from '../../../../utility/mixin';

const StyledFooter = styled.footer`
  display: none;
  margin: 120px 0 50px;
  text-align: center;

  > div {
    &:nth-of-type(1) {
      ${border('4px', true)};
      width: 600px;
      margin: 0 auto 40px;

      div {
        ${border()};

        &:first-of-type {
          display: flex;

          div {
            display: flex;
            flex-direction: column;
            justify-content: center;
            flex-basis: 30%;

            &:not(:first-of-type) {
              margin: 0 0 0 2px;
            }

            &:nth-of-type(2) {
              flex-basis: 40%;
              padding: 10px;

              p {
                margin: 0 0 5px;
              }
            }
          }
        }

        &:last-of-type {
          margin: 2px 0 0;
          padding: 5px;
        }
      }
    }

    &:nth-of-type(2) {
      display: flex;
      justify-content: center;
      align-items: center;
      height: 85px;
      margin: 0 0 20px;
      overflow: hidden;

      img {
        position: relative;

        &:first-of-type,
        &:last-of-type {
          top: 8px;
          height: 70px;
        }

        &:nth-of-type(2) {
          top: 25px;
          margin: 0 30px;
        }
      }
    }

    &:nth-of-type(3) img {
      height: 36px;
      object-fit: cover;
    }
  }
`;

export default StyledFooter;
