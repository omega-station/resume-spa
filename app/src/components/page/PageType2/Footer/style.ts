import styled from '@emotion/styled';
import { breakpoint } from '../../../../utility/constant';
import { border } from '../../../../utility/mixin';

const StyledFooter = styled.footer`
  margin: 120px 0 50px;
  text-align: center;

  > div {
    &:nth-of-type(1) {
      ${border('4px', true)};
      width: 100%;
      max-width: 300px;
      margin: 0 auto 40px;

      @media (min-width: ${breakpoint.mobile[3]}) {
        max-width: 680px;
      }

      div {
        ${border()};

        &:first-of-type {
          display: flex;
          flex-direction: column;

          div {
            display: flex;
            flex-direction: column;
            justify-content: center;
            flex-basis: 30%;

            &:not(:first-of-type) {
              margin: 2px 0 0;
            }

            &:nth-of-type(2) {
              flex-basis: 40%;
              padding: 10px;

              p {
                margin: 0 0 5px;
              }
            }
          }

          @media (min-width: ${breakpoint.mobile[3]}) {
            flex-direction: row;

            div:not(:first-of-type) {
              margin: 0 0 0 2px;
            }
          }
        }

        &:last-of-type {
          margin: 2px 0 0;
          padding: 5px;
        }
      }
    }

    &:nth-of-type(2) {
      display: flex;
      flex-direction: column;
      justify-content: center;
      align-items: center;

      img {
        position: relative;

        &:first-of-type,
        &:last-of-type {
          height: 70px;
        }

        &:first-of-type {
          top: 20px;
        }

        &:last-of-type {
          top: -50px;
        }
      }

      @media (min-width: ${breakpoint.mobile[3]}) {
        flex-direction: row;
        height: 85px;
        margin: 0 0 40px;
        overflow: hidden;

        img {
          &:first-of-type,
          &:last-of-type {
            top: 8px;
          }

          &:nth-of-type(2) {
            top: 25px;
            margin: 0 40px;
          }
        }
      }
    }

    &:nth-of-type(3) img {
      height: 36px;
      object-fit: cover;
    }
  }
`;

export default StyledFooter;
