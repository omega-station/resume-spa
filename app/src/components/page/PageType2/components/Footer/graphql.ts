import gql from 'graphql-tag';

export type gqlAside = any;
export type gqlResume = any;

export const GQL_QUERY = gql`
  query {
    options {
      pageContent {
        pagetype2ImagesFooter0 {
          mediaItemUrl
          sizes(size: LARGE)
          srcSet(size: LARGE)
          title
        }
        pagetype2ImagesFooter1 {
          mediaItemUrl
          sizes(size: LARGE)
          srcSet(size: LARGE)
          title
        }
        pagetype2ImagesFooter2 {
          mediaItemUrl
          sizes(size: LARGE)
          srcSet(size: LARGE)
          title
        }
      }
    }
  }
`;
