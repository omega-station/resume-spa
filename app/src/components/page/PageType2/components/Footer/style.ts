import styled from '@emotion/styled';
import { viewport } from '../../../../../utility/constant';
import { border } from '../../../../../utility/mixin';

const StyledFooter = styled.footer`
  margin: 120px 0 50px;
  text-align: center;

  > div {
    &:nth-of-type(1) {
      ${border(4, true)};
      width: 100%;
      max-width: 300px;
      margin: 0 auto 40px;

      @media (min-width: ${viewport.mobile[3]}) {
        max-width: 680px;
      }

      div {
        ${border()};

        &:first-of-type {
          display: flex;
          flex-direction: column;

          div {
            display: flex;
            flex-direction: column;
            justify-content: center;

            &:not(:first-of-type) {
              margin: 2px 0 0;
            }

            &:nth-of-type(2) {
              padding: 10px;

              p {
                margin: 0 0 5px;
              }
            }
          }

          @media (min-width: ${viewport.mobile[3]}) {
            flex-direction: row;

            div {
              width: 30%;

              &:nth-of-type(2) {
                width: 40%;
              }

              &:not(:first-of-type) {
                margin: 0 0 0 2px;
              }

              img {
                display: block;
                max-width: 300px;
                max-height: 200px;
                width: auto;
                height: auto;
                object-fit: cover;
              }
            }
          }
        }

        &:last-of-type {
          margin: 2px 0 0;
          padding: 5px;
        }
      }
    }

    &:nth-of-type(2) {
      display: flex;
      flex-direction: column;
      justify-content: center;
      align-items: center;

      img {
        position: relative;

        &:first-of-type,
        &:last-of-type {
          height: 70px;
        }

        &:first-of-type {
          top: 20px;
        }

        &:last-of-type {
          top: -50px;
        }
      }

      @media (min-width: ${viewport.mobile[3]}) {
        flex-direction: row;
        height: 85px;
        margin: 0 0 40px;
        overflow: hidden;

        img {
          &:first-of-type,
          &:last-of-type {
            top: 8px;
          }

          &:nth-of-type(2) {
            top: 25px;
            margin: 0 40px;
          }
        }
      }
    }

    &:nth-of-type(3) img {
      height: 36px;
      object-fit: cover;
    }
  }
`;

export default StyledFooter;
