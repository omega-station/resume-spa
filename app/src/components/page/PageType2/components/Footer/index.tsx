import React from 'react';
import { QueryResult, useQuery } from 'react-apollo';
import { getImage, getRandomInt } from '../../../../../utility';
import Error from '../../../../core/Error';
import Loading from '../../../../core/Loading';
import { GQL_QUERY } from './graphql';
import StyledFooter from './style';

const Footer = (): JSX.Element => {
  const { data, loading, error }: QueryResult = useQuery(GQL_QUERY);

  if (loading) return <Loading />;
  if (error) return <Error />;

  const images: any = data.options.pageContent.pagetype2Images;

  const url = 'https://en.wikipedia.org/wiki/Webring';
  const counter = (
    <div>
      <a href="https://www.hitwebcounter.com/" target="_blank" rel="noopener noreferrer">
        <img src="https://hitwebcounter.com/counter/counter.php?page=7200063&style=0038&nbdigits=7&type=page&initCount=325145" alt="hitwebcounter" />
      </a>
    </div>
  );

  return (
    <StyledFooter>
      <div>
        <div>
          <div>{getImage(images[getRandomInt(3, images.length - 1)])}</div>
          <div>
            <p>
              This{' '}
              <a href={url} target="_blank" rel="noopener noreferrer">
                Amazing Front-end Developer's Ring
              </a>{' '}
              site is owned by{' '}
              <a href={url} target="_blank" rel="noopener noreferrer">
                Paul Kevin Koehler
              </a>
              .
            </p>
            <p>
              Want to join the{' '}
              <a href={url} target="_blank" rel="noopener noreferrer">
                Amazing Front-end Developer's Ring
              </a>
              ?
            </p>
          </div>
          <div>{getImage(images[getRandomInt(3, images.length - 1)])}</div>
        </div>
        <div>
          [
          <a href={url} target="_blank" rel="noopener noreferrer">
            Prev
          </a>
          ] [
          <a href={url} target="_blank" rel="noopener noreferrer">
            Next
          </a>
          ] [
          <a href={url} target="_blank" rel="noopener noreferrer">
            Random
          </a>
          ] [
          <a href={url} target="_blank" rel="noopener noreferrer">
            List
          </a>
          ]
        </div>
      </div>
      <div>
        {getImage(images[1])}
        {getImage(images[2])}
        {getImage(images[1])}
      </div>
      {counter}
    </StyledFooter>
  );
};

export default Footer;
