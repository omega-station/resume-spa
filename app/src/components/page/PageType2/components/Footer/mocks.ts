import { GQL_QUERY } from './graphql';

const mocks = [
  {
    request: {
      query: GQL_QUERY,
    },
    result: {
      data: {
        options: {
          pageContent: {
            pagetype2Images: [
              {
                mediaItemUrl: 'http://dev.resume-wp.omegastation.ca/wp-content/uploads/lemming.gif',
                sizes: null,
                srcSet: null,
                title: 'Lemming',
              },
              {
                mediaItemUrl: 'http://dev.resume-wp.omegastation.ca/wp-content/uploads/uc-main.gif',
                sizes: null,
                srcSet: null,
                title: 'Under Construction',
              },
              {
                mediaItemUrl: 'http://dev.resume-wp.omegastation.ca/wp-content/uploads/uc-digger.gif',
                sizes: null,
                srcSet: null,
                title: 'Digger',
              },
            ],
          },
        },
      },
    },
  },
];

export default mocks;
