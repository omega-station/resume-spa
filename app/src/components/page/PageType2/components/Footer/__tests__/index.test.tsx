import { cleanup, render } from '@testing-library/react';
import React from 'react';
import Footer from '..';

afterEach(cleanup);

describe('Footer', () => {
  it('should render without props', () => {
    const component = render(<Footer />);
    expect(component).toMatchSnapshot();
  });
});
