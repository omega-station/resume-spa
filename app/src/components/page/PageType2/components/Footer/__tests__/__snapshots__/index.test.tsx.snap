// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render without props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <footer
        class="css-mpmvk1"
      >
        <div>
          <div>
            <div>
              <img
                alt=""
                src="coding.webp"
              />
            </div>
            <div>
              <p>
                This
                 
                <a
                  href="https://en.wikipedia.org/wiki/Webring"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Amazing Front-end Developer's Ring
                </a>
                 
                site is owned by
                 
                <a
                  href="https://en.wikipedia.org/wiki/Webring"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Paul Kevin Koehler
                </a>
                .
              </p>
              <p>
                Want to join the
                 
                <a
                  href="https://en.wikipedia.org/wiki/Webring"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Amazing Front-end Developer's Ring
                </a>
                ?
              </p>
            </div>
            <div>
              <img
                alt=""
                src="reading.webp"
              />
            </div>
          </div>
          <div>
            [
            <a
              href="https://en.wikipedia.org/wiki/Webring"
              rel="noopener noreferrer"
              target="_blank"
            >
              Prev
            </a>
            ] [
            <a
              href="https://en.wikipedia.org/wiki/Webring"
              rel="noopener noreferrer"
              target="_blank"
            >
              Next
            </a>
            ] [
            <a
              href="https://en.wikipedia.org/wiki/Webring"
              rel="noopener noreferrer"
              target="_blank"
            >
              Random
            </a>
            ] [
            <a
              href="https://en.wikipedia.org/wiki/Webring"
              rel="noopener noreferrer"
              target="_blank"
            >
              List
            </a>
            ]
          </div>
        </div>
        <div>
          <img
            alt=""
            src="uc-digger.gif"
          />
          <img
            alt=""
            src="uc-main.gif"
          />
          <img
            alt=""
            src="uc-digger.gif"
          />
        </div>
        <div>
          <a
            href="https://www.hitwebcounter.com/"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="hitwebcounter"
              src="https://hitwebcounter.com/counter/counter.php?page=7200063&style=0038&nbdigits=7&type=page&initCount=325145"
            />
          </a>
        </div>
      </footer>
    </div>
  </body>,
  "container": <div>
    <footer
      class="css-mpmvk1"
    >
      <div>
        <div>
          <div>
            <img
              alt=""
              src="coding.webp"
            />
          </div>
          <div>
            <p>
              This
               
              <a
                href="https://en.wikipedia.org/wiki/Webring"
                rel="noopener noreferrer"
                target="_blank"
              >
                Amazing Front-end Developer's Ring
              </a>
               
              site is owned by
               
              <a
                href="https://en.wikipedia.org/wiki/Webring"
                rel="noopener noreferrer"
                target="_blank"
              >
                Paul Kevin Koehler
              </a>
              .
            </p>
            <p>
              Want to join the
               
              <a
                href="https://en.wikipedia.org/wiki/Webring"
                rel="noopener noreferrer"
                target="_blank"
              >
                Amazing Front-end Developer's Ring
              </a>
              ?
            </p>
          </div>
          <div>
            <img
              alt=""
              src="reading.webp"
            />
          </div>
        </div>
        <div>
          [
          <a
            href="https://en.wikipedia.org/wiki/Webring"
            rel="noopener noreferrer"
            target="_blank"
          >
            Prev
          </a>
          ] [
          <a
            href="https://en.wikipedia.org/wiki/Webring"
            rel="noopener noreferrer"
            target="_blank"
          >
            Next
          </a>
          ] [
          <a
            href="https://en.wikipedia.org/wiki/Webring"
            rel="noopener noreferrer"
            target="_blank"
          >
            Random
          </a>
          ] [
          <a
            href="https://en.wikipedia.org/wiki/Webring"
            rel="noopener noreferrer"
            target="_blank"
          >
            List
          </a>
          ]
        </div>
      </div>
      <div>
        <img
          alt=""
          src="uc-digger.gif"
        />
        <img
          alt=""
          src="uc-main.gif"
        />
        <img
          alt=""
          src="uc-digger.gif"
        />
      </div>
      <div>
        <a
          href="https://www.hitwebcounter.com/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="hitwebcounter"
            src="https://hitwebcounter.com/counter/counter.php?page=7200063&style=0038&nbdigits=7&type=page&initCount=325145"
          />
        </a>
      </div>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
